content/4. Deeltaken/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/While/1. While.md

```diff
@@ -0,0 +1,352 @@
+```
+Voer het wachtwoord in:
+Fout wachtwoord probeer opnieuw.
+```
+Als invoer `C#Rocks` is:
+```
+Voer het wachtwoord in:
+Succes! Je bent ingelogd
+```
+
+### Nu jij:
+``` csharp runner
+using System;
+
+// Random wachtwoorden  
+string[] randomStrings = {"C#Rocks", "KotlinRocks", "JavaDoesntRock", "PythonRocks" };  
+  
+// Correcte wachtwoord  
+string password = "C#Rocks";  
+  
+// Random gekozen wachtwoord  
+string input = randomStrings[new Random().Next(randomStrings.Length)];  
+  
+// While-loop om het wachtwoord te blijven controleren.  
+while(){  
+
+	// Random gekozen wachtwoord is onjuist, kies een nieuwe
+    input = randomStrings[new Random().Next(randomStrings.Length)];  
+}
+
+// Print als wachtwoord correct is
+```
+
+> [!info]- Mogelijke uitwerking
+>``` csharp runner
+> using System;
+>
+> // Random wachtwoorden
+> string[] randomStrings = {"C#Rocks", "KotlinRocks", "JavaDoesntRock", "PythonRocks" };
+>        
+> // Correcte wachtwoord
+> string password = "C#Rocks";
+>        
+> // Random gekozen wachtwoord
+> string input = randomStrings[new Random().Next(randomStrings.Length)];
+>        
+> // While-loop om het wachtwoord te blijven controleren.
+> while(input != password)
+> {
+> 	Console.WriteLine($"Gekozen wachtwoord is: {input}");
+> 	Console.WriteLine("Fout wachtwoord probeer opnieuw.");
+> 	input = randomStrings[new Random().Next(randomStrings.Length)];
+> }
+>   
+> // Print als wachtwoord correct is
+> Console.WriteLine($"Gekozen wachtwoord is: {input}");
+> Console.WriteLine("Succes! Je bent ingelogd");
+>```
+
+---
+## Opdracht 5
+Schrijf een programma dat een random waarde van een dobbelsteen (1 t/m 6) kiest. Het programma moet een nieuwe worp blijven doen zolang de waarde van de dobbelsteen niet 6 is. Wanneer 6 gegooid is moet er getoond worden hoeveel worpen er gedaan zijn.
+
+### Specificaties
+* Er wordt een random waarde gekozen (tussen 1 en 6)
+* Als de waarde niet 6 is kies dan een nieuwe waarde
+* Als 6 gekozen is toon dan hoeveel worpen het heeft gekost om de waarde 6 te krijgen
+
+### Verwachte output:
+Als de eerste invoer 5 is:
+```
+Dobbelsteenrol 1: 5
+//Nieuwe waarde (3)
+Dobbelsteenrol 2: 3
+//Nieuwe waarde (6)
+Het duurde 3 worpen om een 6 te gooien.
+```
+Als de invoer 6 is:
+```
+Het duurde 1 worpen om een 6 te gooien.
+```
+
+### Nu jij:
+``` csharp runner
+using System;
+ 
+Random rand = new Random();
+int diceValue;
+int count = 0;
+
+// Simuleer dobbelsteenrollen tot een 6 wordt gegooid
+do
+{
+  diceValue = rand.Next(1, 7); // Dobbelsteenwaarde tussen 1 en 6
+
+} while ();
+
+// Toon resultaat
+
+```
+
+> [!info]- Mogelijke uitwerking
+> ``` csharp
+> using System;
+> 
+> Random rand = new Random();
+> int diceValue;
+> int count = 0;
+> 
+> // Simuleer dobbelsteenrollen tot een 6 wordt gegooid
+> do
+> {
+>   diceValue = rand.Next(1, 7); // Dobbelsteenwaarde tussen 1 en 6
+>   count++;
+>   Console.WriteLine($"Dobbelsteenrol {count}: {roll}");
+> } while (roll != 6);
```

Comment:
Opmerking (ook betreffende r346): je noemt het nu 'roll', wat ik nu voor het eerst in de opgave lees. Bedoel je hier diceValue?